<a href="https://colab.research.google.com/github/map72ru/linag/blob/main/%D0%94%D0%975.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. Найти с помощью NumPy SVD для матрицы

$$\begin{pmatrix}
1 & 2 & 0\\ 
0 & 0 & 5\\ 
3 & -4 & 2\\ 
1 & 6 & 5\\ 
0 & 1 & 0
\end{pmatrix}.$$

In [1]:
import numpy as nm

In [6]:
A = nm.array([[1,2,0],[0,0,5],[3,-4,2],[1,6,5],[0,1,0]])
U,D,V = nm.linalg.svd(A)
print(f'U={U}')
print(f'D={D}')
print(f'Vt={V}')


U=[[ 0.17056501  0.15680918 -0.53077508 -0.79905375 -0.16158397]
 [ 0.39287016 -0.52933945  0.6134793  -0.43375771  0.03082495]
 [-0.14366152 -0.82449256 -0.52379105  0.14049848  0.07400343]
 [ 0.88843702  0.06074346 -0.24655277  0.37755832 -0.06042632]
 [ 0.08125046  0.10831843 -0.08231425 -0.10524851  0.98173958]]
D=[8.82486885 6.14060608 2.53271528]
Vt=[[ 0.07116451  0.71702467  0.69340553]
 [-0.36737824  0.66514082 -0.65009301]
 [-0.92734505 -0.20847855  0.31075368]]


Для матрицы из предыдущего задания найти:

а) евклидову норму;

б) норму Фробениуса.

Норма Евклида равна $\mu_1$ сингулярного разложения. Сл-но: $\left \| A \right \|_{E}=8.82486885$

Норма Фрабениуса равна

$$\left \| A \right \|_{F}=\sqrt{\sum_{k=1}^{r}\mu_{k}^{2}}.$$

Следовательно:

$$\left \| A \right \|_{F}=\sqrt{8.82486885^2+6.14060608^2+2.53271528^2}=11.045361014425$$

In [7]:
nm.linalg.norm(A)

11.045361017187261